components/layout_2020/flow/inline.rs

```diff
@@ -284,7 +289,136 @@ impl<'box_tree> PartialInlineBoxFragment<'box_tree> {
 }
 
 impl TextRun {
-    fn layout(&self, _ifc: &mut InlineFormattingContextState) {
-        // TODO
+    fn layout(&self, layout_context: &LayoutContext, ifc: &mut InlineFormattingContextState) {
+        use gfx::font::{ShapingFlags, ShapingOptions};
+        use style::computed_values::text_rendering::T as TextRendering;
+        use style::computed_values::word_break::T as WordBreak;
+        use style::values::generics::text::LineHeight;
+
+        let font_style = self.parent_style.clone_font();
+        let inherited_text_style = self.parent_style.get_inherited_text();
+        let letter_spacing = Some(inherited_text_style.letter_spacing.0)
+            .filter(|spacing| spacing.px() != 0.)
```

Comment:
Nit: `Some(…).filter(…)` is clever but this looks like it might just as well be an `if` / `else`?